### PR TITLE
Bug 1809861: add extract image init container failure test

### PR DESCRIPTION
**Pull request by gabemontero**
/hold

until https://github.com/openshift/builder/pull/142 merges new test should fail
we will initially want to make sure the new test fails before https://github.com/openshift/builder/pull/142 merges

@openshift/openshift-team-developer-experience 

/assign @bparees 

note as the code path in openshift/builder is exactly the same for docker and s2i strategies, I chose to not add tests for both

also, an easily producible  `ManageDockerfile` failure did not come to mind

### CODE REVIEW

File: test/extended/builds/failure_status.go [hunk 1]
```diff
@@ -85,20 +87,39 @@ var _ = g.Describe("[sig-builds][Feature:Builds][Slow] update failure status", f
...
+
+				br, err := exutil.StartBuildAndWait(oc, "statusfail-fetchsourcedocker", "--build-loglevel=5")
+				o.Expect(err).NotTo(o.HaveOccurred())
+				br.AssertFailure()
+				br.DumpLogs()
+
+				build, err := oc.BuildClient().BuildV1().Builds(oc.Namespace()).Get(br.Build.Name, metav1.GetOptions{})
+				o.Expect(err).NotTo(o.HaveOccurred())
+				o.Expect(build.Status.Reason).To(o.Equal(buildv1.StatusReasonFetchImageContentFailed))
+				o.Expect(build.Status.Message).To(o.Equal(StatusMessageFetchSourceFailed))
```
Comment (bparees) [inline test/extended/builds/failure_status.go]:
```suggestion
				o.Expect(build.Status.Message).To(o.Equal(StatusMessageFetchImageContentFailed))
```

?

File: test/extended/builds/failure_status.go [hunk 2]
```diff
@@ -85,20 +87,39 @@ var _ = g.Describe("[sig-builds][Feature:Builds][Slow] update failure status", f
...
+				br, err := exutil.StartBuildAndWait(oc, "statusfail-fetchimagecontent", "--build-loglevel=5")
 				o.Expect(err).NotTo(o.HaveOccurred())
 				br.AssertFailure()
 				br.DumpLogs()
 
 				build, err := oc.BuildClient().BuildV1().Builds(oc.Namespace()).Get(br.Build.Name, metav1.GetOptions{})
 				o.Expect(err).NotTo(o.HaveOccurred())
 				o.Expect(build.Status.Reason).To(o.Equal(buildv1.StatusReasonFetchSourceFailed))
-				o.Expect(build.Status.Message).To(o.Equal(StatusMessageFetchSourceFailed))
+				o.Expect(build.Status.Message).To(o.Equal(StatusMessageFetchImageContentFailed))
```
Comment (bparees) [inline test/extended/builds/failure_status.go]: ?

File: test/extended/builds/failure_status.go [hunk 3]
```diff
@@ -85,20 +87,39 @@ var _ = g.Describe("[sig-builds][Feature:Builds][Slow] update failure status", f
...
+			})
+		})
+
 		g.Describe("Build status Docker fetch source failure", func() {
 			g.It("should contain the Docker build fetch source failure reason and message", func() {
 				err := oc.Run("create").Args("-f", fetchDockerSrc).Execute()
 				o.Expect(err).NotTo(o.HaveOccurred())
 
-				br, err := exutil.StartBuildAndWait(oc, "statusfail-fetchsourcedocker", "--build-loglevel=5")
+				br, err := exutil.StartBuildAndWait(oc, "statusfail-fetchimagecontent", "--build-loglevel=5")
```
Comment (bparees) [inline test/extended/builds/failure_status.go]: ?

File: test/extended/builds/failure_status.go [hunk 4]
```diff
@@ -85,20 +87,39 @@ var _ = g.Describe("[sig-builds][Feature:Builds][Slow] update failure status", f
 			})
 		})
 
+		g.Describe("Build status Docker fetch image content failure", func() {
+			g.It("should contain the Docker build fetch image content reason and message", func() {
+				err := oc.Run("create").Args("-f", fetchDockerImg).Execute()
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				br, err := exutil.StartBuildAndWait(oc, "statusfail-fetchsourcedocker", "--build-loglevel=5")
```
Comment (bparees) [inline test/extended/builds/failure_status.go]:
```suggestion
				br, err := exutil.StartBuildAndWait(oc, "statusfail-fetchimagecontent", "--build-loglevel=5")
```

File: test/extended/testdata/builds/statusfail-fetchimagecontentdocker.yaml [hunk 5]
```diff
@@ -0,0 +1,22 @@
+kind: BuildConfig
+apiVersion: v1
+metadata:
+  name: statusfail-fetchsourcedocker
```
Comment (bparees) [inline test/extended/testdata/builds/statusfail-fetchimagecontentdocker.yaml]:
```suggestion
  name: statusfail-fetchimagecontent
```

**Comment by gabemontero** [comment]
> looks like a lot of copy/paste/mutate errors in here.

sigh ... yes there are ... thanks for the commit suggestions, but I may just do it on my end 

the yaml in particular I swore I changed ... maybe I got them confused with logsnippet changes

I think I've reached critical mass on how many bugs I can work concurrently ;-)

apologies for the churn

**Comment by gabemontero** [comment]
ok the `failure_status.go` diff looks sane now @bparees

**Comment by gabemontero** [comment]
Our new test case failed as expected in e2e-gcp-builds:
```

[sig-builds][Feature:Builds][Slow]  update failure status  Build status Docker fetch image content failure  should contain the Docker build fetch image content reason and message  [Suite:openshift] expand_less | 22s
-- | --
fail [github.com/openshift/origin/test/extended/builds/failure_status.go:102]: Expected     <v1.StatusReason>: GenericBuildFailed to equal     <v1.StatusReason>: FetchImageContentFailed

```

**Pull request merged**